idl/config.x

```diff
@@ -0,0 +1,63 @@
+%#include "common.x"
+
+namespace mazzaroth
+{
+  // ContractChannelConfig stores contract channel configuration in state and is 
+  // accessible through host contract host functions
+  struct ContractChannelConfig
+  {
+    // Unique channel id which cannot be changed
+    ID channelID;
+    // Contract binary bytes
+    opaque contract<>;
+    // Version number of the contract, specified by owner
+    string version;
+    // Public Key ID of the channel owner. Only owner can change this to transfer ownership of channel
+    ID owner;
+    // Human readable channel name
+    string channelName;
+    // Public Keys of IDs approved by owner able to modify channel
+    ID admins<>;
+  };
+
+  // GovernanceConfig stores governance information about a channel 
+  struct GovernanceConfig
+  {
+    // Max number of transactions to store in a block
+    unsigned hyper maxBlockSize;
+    // Consensus type being used
+    ConsensusConfigType consensus;
+    // Permissioning configurations
+    Permissioning permissioning;
+  }
+
+  // Consensus type
+  enum ConsensusConfigType
+  {
+    NONE = 0,
+    PBFT = 1
+  };
+
+  // Permissioning types
+  enum PermissioningType
+  {
+    PUBLIC = 0,
```

Comment:
Didn't we have a 3rd category of Permissioned? Maybe a bad name, but I think we need to account for the concept.

Reply:
Yes, I have updated this enum.